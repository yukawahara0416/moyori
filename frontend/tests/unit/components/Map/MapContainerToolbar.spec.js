import { shallowMount } from '@vue/test-utils'
import Component from '@/components/Map/MapContainerToolbar.vue'

let wrapper

beforeEach(() => {
  wrapper = shallowMount(Component, {})
})

afterEach(() => {
  wrapper.destroy()
})

describe('v-on', () => {
  it('textSearch', () => {
    wrapper.setMethods({ textSearch: jest.fn() })
    wrapper.vm.$emit('textSearch')
    expect(wrapper.emitted().textSearch).toBeTruthy()
  })
  it('panToLocation', () => {
    wrapper.setMethods({ panToLocation: jest.fn() })
    wrapper.vm.$emit('panToLocation')
    expect(wrapper.emitted().panToLocation).toBeTruthy()
  })
  it('nearbySearch', () => {
    wrapper.setMethods({ nearbySearch: jest.fn() })
    wrapper.vm.$emit('nearbySearch')
    expect(wrapper.emitted().nearbySearch).toBeTruthy()
  })
})

describe('emit', () => {
  it('textSearch', () => {
    wrapper.vm.textSearch()
    expect(wrapper.emitted('text-search')).toBeTruthy()
  })
  it('panToLocation', () => {
    wrapper.vm.panToLocation()
    expect(wrapper.emitted('panto-location')).toBeTruthy()
  })
  it('nearbySearch', () => {
    wrapper.vm.nearbySearch()
    expect(wrapper.emitted('nearby-search')).toBeTruthy()
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
