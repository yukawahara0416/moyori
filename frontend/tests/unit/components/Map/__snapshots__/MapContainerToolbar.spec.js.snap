// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<v-toolbar-stub
  extensionheight="48"
  src=""
  style="position: absolute; left: 10px; top: 10px; z-index: 2;"
  tag="header"
  tile="true"
>
  <map-container-toolbar-text-form-stub />
   
  <map-container-toolbar-panto-button-stub />
   
  <map-container-toolbar-nearby-button-stub />
</v-toolbar-stub>
`;
